avogadro/io/trrformat.cpp

```diff
@@ -0,0 +1,446 @@
+/******************************************************************************
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2018 Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+
+#include "trrformat.h"
+#include "struct.h"
+
+#include <avogadro/core/elements.h>
+#include <avogadro/core/molecule.h>
+#include <avogadro/core/unitcell.h>
+#include <avogadro/core/utilities.h>
+#include <avogadro/core/vector.h>
+
+#include <iomanip>
+#include <istream>
+#include <ostream>
+#include <sstream>
+#include <string>
+
+using std::endl;
+using std::getline;
+using std::map;
+using std::pair;
+using std::string;
+using std::to_string;
+using std::vector;
+
+namespace Avogadro {
+namespace Io {
+
+using Core::Array;
+using Core::Atom;
+using Core::Elements;
+using Core::lexicalCast;
+using Core::Molecule;
+using Core::split;
+using Core::trimmed;
+using Core::UnitCell;
+
+#ifndef _WIN32
+using std::isalpha;
+#endif
+
+#define GROMACS_MAGIC 1993
+#define DIM 3
+#define NM_TO_ANGSTROM 10.0
+string TRR_VERSION = "GMX_trn_file";
+string HEAD_ITEMS[] = { "ir_size",   "e_size",   "box_size", "vir_size",
+                        "pres_size", "top_size", "sym_size", "x_size",
+                        "v_size",    "f_size",   "natoms",   "step",
+                        "nre",       "time",     "lambda" };
+
+int swap_integer(int inp)
```

Comment:
Should stick to the same naming convention, swapInteger, rather than STL style swap_integer